qa/rpc-tests/wallet_accounts.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Zcash developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+from test_framework.authproxy import JSONRPCException
+from test_framework.mininode import COIN
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_raises_message,
+    get_coinbase_address,
+    start_nodes,
+    wait_and_assert_operationid_status,
+)
+
+from decimal import Decimal
+
+# Test wallet accounts behaviour
+class WalletAccountsTest(BitcoinTestFramework):
+    def setup_nodes(self):
+        return start_nodes(self.num_nodes, self.options.tmpdir, [[
+            '-experimentalfeatures',
+            '-orchardwallet',
+        ]] * self.num_nodes)
+
+    def check_receiver_types(self, ua, expected):
+        actual = self.nodes[0].z_listunifiedreceivers(ua)
+        assert_equal(set(expected), set(actual))
+
+    # Check we only have balances in the expected pools.
+    # Remember that empty pools are omitted from the output.
+    def check_account_balance(self, account, expected, minconf=None):
+        if minconf is None:
+            actual = self.nodes[0].z_getbalanceforaccount(account)
+        else:
+            actual = self.nodes[0].z_getbalanceforaccount(account, minconf)
+        assert_equal(set(expected), set(actual['pools']))
+        for pool in expected:
+            assert_equal(expected[pool] * COIN, actual['pools'][pool]['valueZat'])
+        assert_equal(actual['minimum_confirmations'], 1 if minconf is None else minconf)
+
+    # Check we only have balances in the expected pools.
+    # Remember that empty pools are omitted from the output.
+    def check_address_balance(self, address, expected, minconf=None):
+        if minconf is None:
+            actual = self.nodes[0].z_getbalanceforaddress(address)
+        else:
+            actual = self.nodes[0].z_getbalanceforaddress(address, minconf)
+        assert_equal(set(expected), set(actual['pools']))
+        for pool in expected:
+            assert_equal(expected[pool] * COIN, actual['pools'][pool]['valueZat'])
+        assert_equal(actual['minimum_confirmations'], 1 if minconf is None else minconf)
+
+    def check_balance(self, account, address, expected, minconf=None):
+        self.check_account_balance(account, expected, minconf)
+        self.check_address_balance(address, expected, minconf)
```

Comment:
This code is almost-duplicated from `wallet_z_sendmany.py` added in #5469 (this version fixes the `node` argument to `self.nodes[0]`). Either apply the same simplification as in https://github.com/zcash/zcash/pull/5469/commits/cf271473ebf75fa367419454e6a0403ca4eb0324, or factor out the duplicated code into a utility function.

```suggestion
    def _check_balance_for_rpc(self, rpcmethod, node, account, expected, minconf):
        rpc = getattr(self.nodes[node], rpcmethod)
        actual = rpc(account) if minconf is None else rpc(account, minconf)
        assert_equal(set(expected), set(actual['pools']))
        for pool in expected:
            assert_equal(expected[pool] * COIN, actual['pools'][pool]['valueZat'])
        assert_equal(actual['minimum_confirmations'], 1 if minconf is None else minconf)

    def check_balance(self, account, address, expected, minconf=None):
        self._check_balance_for_rpc('z_getbalanceforaccount', self.nodes[0], account, expected, minconf)
        self._check_balance_for_rpc('z_getbalanceforaddress', self.nodes[0], address, expected, minconf)
```